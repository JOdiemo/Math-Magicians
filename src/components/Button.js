import PropTypes from 'prop-types';
import calculate from './logic/calculate';

const Button = ({
  obj, setObj, value, classList,

}) => {
  const handleInput = (e) => {
    setObj(calculate(obj, e.target.innerText));
  };

  return (
    <button
      type="button"
      className={classList}
      onClick={(e) => {
        handleInput(e);
      }}
    >
      {value}
    </button>
  );
};

Button.propTypes = {
  obj: PropTypes.string.isRequired,
  setObj: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  classList: PropTypes.string.isRequired,
};

export default Button;
