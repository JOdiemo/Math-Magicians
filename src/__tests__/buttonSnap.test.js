import React from 'react';
import renderer from 'react-test-renderer';
// import { BrowserRouter as Router } from 'react-router-dom';
import Button from '../components/Button';

describe('Button component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Button />);
    expect(tree).toMatchSnapshot();
  });
});