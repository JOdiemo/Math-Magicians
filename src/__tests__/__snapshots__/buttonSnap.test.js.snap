// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`Navbar component matches the snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;
